src/mongo/db/pipeline/expression.cpp

```diff
@@ -1729,9 +1729,18 @@ intrusive_ptr<ExpressionObject> ExpressionObject::parse(
 }
 
 intrusive_ptr<Expression> ExpressionObject::optimize() {
+    size_t countConstants = 0;
```

Comment:
You don't really need to count them all, do you? A simple boolean should suffice:
`bool allValuesConstant = true`